packages/toolpad-app/src/server/data.ts

```diff
@@ -175,17 +176,36 @@ function createDefaultDom(): appDom.AppDom {
   return dom;
 }
 
-export interface CreateAppOptions {
-  dom?: appDom.AppDom | null;
-}
-
-export async function createApp(name: string, opts: CreateAppOptions = {}): Promise<prisma.App> {
+export type CreateAppOptions =
+  | {
+      type: 'dom';
+      dom?: appDom.AppDom | null;
+    }
+  | {
+      type: 'template';
+      name: AppTemplateName;
+    };
+
+export async function createApp(
+  name: string,
+  opts: CreateAppOptions = { type: 'dom' },
```

Comment:
```ts
export interface CreateAppOptions {
  from?:
    | {
        kind: 'dom';
        dom: appDom.AppDom;
      }
    | {
        kind: 'template';
        id: string;
      };
}
```

Represents the options better. `from: undefined` represents "just create an empty app". and `{ kind: 'dom' }` becomes illegal. The type system helps you to not accidentally forget to pass a dom.

I also prefer to move more towards the `kind` nomenclature instead of `type` for discriminated unions.

Reply:
sounds good! let's do it like that then